### ISSUE TITLE
Fatal PHP error upon activation/configuration

### ISSUE THREAD

**Issue by seedcity** [taxjar/taxjar-woocommerce-plugin#258]
We installed/activated Taxjar on our WooCommerce site and only after enabling the configuration settings (check boxes) to actually turn on the tax calculations on the front end did things start to break. Now, while things on the front end seem to be working correctly, as in the tax is displaying and calculating correctly at checkout, it’s simply the admin area that we’re getting the ‘fatal error’ that is breaking other features and plugins.

**Here’s the error:**

```
Fatal error: Uncaught Error: TaxJar\Order_Meta_Box::get_order_tax_calculation_metadata(): Argument #1 ($order) must be of type WC_Order, bool given, called in /home/name/www/site-name.bak/wp-content/plugins/taxjar-simplified-taxes-for-woocommerce/includes/admin/class-order-meta-box.php on line 32
in /home/name/www/site-name.bak/wp-content/plugins/taxjar-simplified-taxes-for-woocommerce/includes/admin/class-order-meta-box.php on line 45

Call stack:

TaxJar\Order_Meta_Box::get_order_tax_calculation_metadata()
wp-content/plugins/taxjar-simplified-taxes-for-woocommerce/includes/admin/class-order-meta-box.php:32
TaxJar\Order_Meta_Box::output()
wp-admin/includes/template.php:1456
do_meta_boxes()
wp-admin/edit-form-advanced.php:723
require()
wp-admin/post.php:206
```

<img width="820" alt="Screenshot 2024-08-07 at 10 47 49 AM" src="https://github.com/user-attachments/assets/d03ae61a-68e0-40c0-a91c-656529875735">

Can anyone help? I can't seem to reach anyone at Taxjar.

**Comment by ahumulescu** [comment]
Hi @seedcity,

Thank you for raising this issue!

Unfortunately, we aren't able to reproduce it. Is that error displayed in some specific admin pages? Or is it displayed for all of them?

I have created a [PR](https://github.com/taxjar/taxjar-woocommerce-plugin/pull/260) with a potential fix for it. Would it be possible to add the fix to your website and check if the error is still displayed, please?

**Comment by seedcity** [comment]
Hi there,

To answer your question - from what I can see so far, that error is specifically displayed inside the product pages (back end admin). It is not on all admin pages.

And for further clarification, for what it's worth -

Installing and activating TaxJar caused no issues.
Only after enabling/checking 'sales tax calculation' did the error appear. And any combination of the enablement of those other check boxes in the configuration settings doesn't change anything. It could be one or all of them, and the error will still appear.

**Comment by ahumulescu** [comment]
Hello,

Yes, I can reproduce it in the product pages, although it only displays that error in the TaxJar tab. Is it impacting your website in any other negative way than this?

The mentioned PR should fix your issue. Can you try to add it to your website, please?

**Comment by seedcity** [comment]
You are correct - that error does only appear to be displayed in the TaxJar tab within the admin product pages. However, it is indeed impacting the website in negative ways in that it's now allowing us to fully edit/configure our products and it's also breaking other plugins that we have which relate to products (a bundle buying plugin, a bulk buying plugin, etc)

**Comment by ahumulescu** [comment]
Hello,

A new version, 4.2.3, that includes the mentioned fix was released.

Can you give it a try and let us know if it does work as expected now, please?
